tests/integration/devfile/utils/utils.go

```diff
@@ -585,3 +586,32 @@ func DeleteLocalConfig(args ...string) {
 	}
 	helper.MatchAllInOutput(output, expectedOutput)
 }
+
+// VerifyCatalogListComponent verifies component exists in both S2I Component list and Devfile Component list
+func VerifyCatalogListComponent(output string) error {
+
+	componentName := "nodejs"
```

Comment:
pass it as a argument in the calling function.

Reply:
`componentName` is not connecting to the spec anywhere https://github.com/openshift/odo/pull/3715/files#diff-318517042700080474192fc7dc04e456R133 and we can check for any component inside the json. So I think its better to have it inside function inspite of unnecessarily adding it to the test.

Reply:
https://github.com/openshift/odo/pull/3715/files#diff-318517042700080474192fc7dc04e456R133 and more importantly is not a good idea to hardcode things in a library function.

Reply:
Yes, its checking the component inside the spec and that's what `VerifyCatalogListComponent()` is doing. So separate function has meant for that purpose only thats why componentName lies there. AFA hardcoding things are concerned so its just a variable name thats needs to be assigned some value right?

Reply:
Think in a generic way, for example tomorrow i want java component name to be verified the then the library function should be flexible to verify it.

Reply:
Verifies the catalog list output along with the component name
```
func VerifyCatalogListComponent(output, componentName string) error {
}
```